### PR TITLE
Migrate MinAggregator integration tests to AggregatorTestCase

### PR DESCRIPTION
**Pull request by polyfractal**
Part of https://github.com/elastic/elasticsearch/issues/42893, this migrates the min aggregator integration tests over to unit tests in AggregatorTestCase.  

While I was messing around with the unit tests, I refactored some of the existing unit tests to use the standardized `searchAndReduce` rather than bespoke version that was in place before.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/search/aggregations/metrics/MinAggregatorTests.java [hunk 1]
```diff
@@ -43,183 +44,534 @@
...
+            assertEquals(1.0, min.getValue(), 0);
+            assertThat(global.getProperty("min"), equalTo(min));
+            assertThat(global.getProperty("min.value"), equalTo(1.0));
+            assertThat(min.getProperty("value"), equalTo(1.0));
+        }, fieldType);
+    }
 
-        indexReader.close();
-        directory.close();
+    public void testSingleValuedFieldPartiallyUnmapped() throws IOException {
```
Comment (not-napoleon) [inline server/src/test/java/org/elasticsearch/search/aggregations/metrics/MinAggregatorTests.java]: Would be good to try this with a missing value set and make sure that gets picked up correctly too.

**Review by not-napoleon — APPROVED** [review]
LGTM.  There's some other cases we could test, like an incorrect type field or weird missing values, but I don't know that we need to do that right now.  Long term, it'd be good to come up with a "standard battery" of tests we should apply to all aggs, but this is a great step towards standardized tests!

**Comment by polyfractal** [comment]
Added a few more tests for missing, unmapped fields, bad types, bad missing, etc.  Happy to add more if you have particular scenarios not covered. :)

**Review by not-napoleon — APPROVED** [review]
These tests look great.  I can't think of any other cases we urgently need to cover, so let's merge it.

**Pull request merged**